### ISSUE TITLE
Could not load file or assembly ‘System.Xml.XmlSerializer

### ISSUE THREAD

**Issue by NicolasDorier** [hhblaze/DBreeze#37]
On linux I got this exception, I added ref to nuget package `System.Xml.XmlSerializer, Version=4.0.11.0` and it seems to have done the trick... unsure why it happens though, your code has no dependency on XmlSerializer.

```
System.IO.FileNotFoundException: Could not load file or assembly ‘System.Xml.XmlSerializer, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a’. The system cannot find the file specified.
File name: ‘System.Xml.XmlSerializer, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a’
   at DBreeze.Transactions.TransactionsCoordinator.Commit(Int32 transactionThreadId)
   at DBreeze.Transactions.Transaction.Commit()
   at ElementsExplorer.Repository.MarkAsUsed(KeyInformation info) in /home/zwoc/workspace/ElementsExplorer/ElementsExplorer/Repository.cs:line 203
   at ElementsExplorer.Controllers.MainController.<Sync>d__5.MoveNext() in /home/zwoc/workspace/ElementsExplorer/ElementsExplorer/Controllers/MainController.cs:line 46
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Mvc.Internal.ObjectMethodExecutor.<CastToObject>d__38`1.MoveNext()
```

**Comment by hhblaze** [comment]
From nuget?

**Comment by NicolasDorier** [comment]
yes

**Comment by NicolasDorier** [comment]
DBReeze version 1.87.0

**Comment by hhblaze** [comment]
yes, would be nice to understand from which nuget folder your DBreeze assembly was taken, can you take a look? NetStandard1.6? Also which dev. environment do you use?

**Comment by NicolasDorier** [comment]
I don't have access to the machine where it faulted now. On my own machine it takes `1.87.0\lib\netcoreapp1.0\DBreeze.dll`.

**Comment by NicolasDorier** [comment]
visual studio windows

**Comment by NicolasDorier** [comment]
More info about the environment where it crashed

```
lefty :: ~ » dotnet --version
1.0.4
lefty :: ~ » uname -a
Linux lefty 4.10.0-26-generic #30-Ubuntu SMP Tue Jun 27 09:30:12 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
lefty :: ~ » lsb_release -a
No LSB modules are available.
Distributor ID:    Ubuntu
Description:    Ubuntu 17.04
Release:    17.04
Codename:    zesty
lefty :: ~ »
```

**Comment by hhblaze** [comment]
thanx, will take a look

**Comment by hhblaze** [comment]
New DBreeze 1.88 is released and also in nuget (looks like I was waiting for you to release 👍 )

**Comment by NicolasDorier** [comment]
@hhblaze you mean the bug I reported is fixed ?

**Comment by hhblaze** [comment]
Has to be fixed, write me if you have troubles - I will reopen it for further investigation.

**Comment by hhblaze** [comment]
May be this someone can point which DLL is used from NuGet, smth like 

![image](https://user-images.githubusercontent.com/486781/29272587-fa32ea46-8100-11e7-828f-6c1b40c5d234.png)

**Comment by NicolasDorier** [comment]
Ok so I could reproduce myself... here is the stange thing:
My tests are working just fine, both using netcoreapp1.0.

`C:\Users\NicolasDorier\.nuget\packages\dbreeze\1.88.0\lib\netcoreapp1.0\DBreeze.dll`

The error comes in my console project oO

**Comment by NicolasDorier** [comment]
Ok so adding  System.Xml.XmlSerializer package fixed the situation.

I am puzzled though, my test project do not reference it, use the same .net runtime, and do not have this problem.
I also see no reason why your code would depends on such assembly

**Comment by hhblaze** [comment]
Yes, it's time to make it.

**Comment by hhblaze** [comment]
I have put in NuGet 1.88.1, 
there is netcoreapp1.1 is available, 
please, test it

**Comment by NicolasDorier** [comment]
I confirm this fixed the situation.

**Comment by NicolasDorier** [comment]
Actually no, same error in another environment.... My PC now works without the Xml stuff this is very wierd

**Comment by NicolasDorier** [comment]
actually it is .NET Core 2.0... I am surprised Azure is already updated to it; it released yestersday.
But your lib support .netstandard 1.4 so why do it not work ?

**Comment by hhblaze** [comment]
The question remains same from which NuGet folder DBreeze is taken for the specified projects (where it does not work) and underlying framework version of this project.

Also we support .NETStandard 1.6 (not 1.4)
![image](https://user-images.githubusercontent.com/486781/29314710-88b5cff4-81bf-11e7-8ae1-a57aa655adc1.png)

Concerning 2.0 - nice, will wait until VS2017 will update its support for it.

**Comment by NicolasDorier** [comment]
@hhblaze VS2017 updated the support. version 15.3.  I don't really know how to find out which dll is used as it is running on a server on which I only have command line available. :(

**Comment by NicolasDorier** [comment]
Ok so it turns out `netcoreapp1.0` was picked.

**Comment by hhblaze** [comment]
But netcoreapp1.1 has System.Xml.XmlSerializer integrated?

![image](https://user-images.githubusercontent.com/486781/29352324-98ee73ce-8266-11e7-980c-25646f3c2eaa.png)

Concerning VS2017 v15.3... my VS doesn't suggest me to be updated (normally does, so, probably I have to wait)

**Comment by NicolasDorier** [comment]
You have to restart vs2017 for the notification to appear.
System.Xml.XmlSerializer was not into my nuget dependencies, to get dbreeze working on my azure VM, I needed to add it.

No problem on my home PC.... this bug is very strange. I can give you a way to reproduce:

Create a ASP.NET core website with DBreeze and Deploy on Azure.

**Comment by hhblaze** [comment]
I think when I get VS2017 15.3 (cause restart doesn't help for now), I will release .NETStandard 2.0, necoreapp 2.0 and will remove dependency in necoreapp  from System.Xml.Serializer at all, like it's done in current .NET standard.

**Comment by hhblaze** [comment]
Released 1.88.2 (when it's available from nuget) both .netcoreapps have no XmlSerializer dependencies

VS 15.3 still not available...

**Comment by hhblaze** [comment]
So, I have update my VS via its installer (automatically update flag didn't appear)
and now with newest Nuget 1.88.3 we got support of .NetCore 2.0 and .NetStandard 2.0
